pkg/koordlet/util/system/cgroup_driver.go

```diff
@@ -150,12 +152,19 @@ var cgroupPathFormatterInSystemd = Formatter{
 				prefix: "cri-containerd-",
 				suffix: ".scope",
 			},
+			{
+				prefix: "crio-",
+				suffix: ".scope",
+			},
 		}
 
 		for i := range patterns {
 			if strings.HasPrefix(basename, patterns[i].prefix) && strings.HasSuffix(basename, patterns[i].suffix) {
 				return basename[len(patterns[i].prefix) : len(basename)-len(patterns[i].suffix)], nil
-			}
+			} 
+		}
+		if strings.HasPrefix(basename, "crio-") {
```

Comment:
why do we need this check

Reply:
Hi zwzhang, thanks for your review. Add this check here for support to pasre container protocol cri-o, for example: cri-o://088b531d2f86e63c6f35c823078 1be16fb0fe5788c14761812d1e8666596e0d.

Reply:
does this check duplicate L162?

Reply:
Hi saintube , 
 This check may not exataly the same as L162. For CRI-O environment, not all the crio protocol process have the ".scope" suffix.Just have the "crio" prefix. 

```

# systemd-cgls -a | grep crio

│     └─17059 grep --color=auto crio
│ ├─crio-conmon-d13050ec5f720a84154e6377ce924a3038e284f0818e4360ed4c08e02449dbb2.scope
│ ├─crio-conmon-b9b0100e6d19edea3cdab408fdd5a41179ac5afa510aee97497b9e10e02b465c.scope
│ ├─crio-conmon-c7d3f699d9101a4298aefbc231d3841a812a243f29961f72a8fb4d4894c0eb31.scope
│ ├─crio.service
│ │ └─6207 /usr/bin/crio
│ ├─crio-conmon-395eb442f16b00dad14eb04ae3b9b24ff14576357ff7fd9c996001c08a0e6b98.scope
│ ├─crio-conmon-26b5f607f2487eb67190b6d401767d1894e3df9d35ed52ac5d104ae2a0881fa9.scope
│ ├─crio-conmon-01008951c16e7620f3ffafe9adf9c5f9c01c7111cfa0f14f81eb3380b2d69460.scope
│ ├─crio-conmon-3469cd41b74f8a3eb39b88d827c3d4263c5dd52a1016f939f926ffe4247d276a.scope
│ ├─crio-conmon-02ba68c0a67bb9d9e3d436daf9a4407212ed8db9b5821bca398923f49db796bd.scope
  │ │ ├─crio-d13050ec5f720a84154e6377ce924a3038e284f0818e4360ed4c08e02449dbb2.scope
  │ │ └─crio-313b25bc1d16516f0e868423bd6d9ebd629f13b566099fff6a242342b2917504
  │ │ ├─crio-5e0a63649249316f641d9151e247201f634c3b1f84ae55b9b3d1cfdc6999d304
  │ │ └─crio-02ba68c0a67bb9d9e3d436daf9a4407212ed8db9b5821bca398923f49db796bd.scope
  │ │ ├─crio-01008951c16e7620f3ffafe9adf9c5f9c01c7111cfa0f14f81eb3380b2d69460.scope
  │ │ └─crio-e12cacaf800b4f7cb6bbdd5c02d5a8a32ca4fd4858b96845beb7dabc6f58b5f1
  │ │ ├─crio-c4c156a7882cea9073000c42194d51aec0e26297762a94f8c153ef0763532763
  │ │ └─crio-26b5f607f2487eb67190b6d401767d1894e3df9d35ed52ac5d104ae2a0881fa9.scope
  │ │ ├─crio-3469cd41b74f8a3eb39b88d827c3d4263c5dd52a1016f939f926ffe4247d276a.scope
  │ │ └─crio-260e7dd9cc5ad3fab2f90f71a244c14113b99619eaaf63ca65fadc1e97f28b82
  │   ├─crio-5b4eda75c6f0f4a122f81d4fa42dd44ababfa440ebe325bb8347bad200e21f48
  │   └─crio-b9b0100e6d19edea3cdab408fdd5a41179ac5afa510aee97497b9e10e02b465c.scope
    │ ├─crio-395eb442f16b00dad14eb04ae3b9b24ff14576357ff7fd9c996001c08a0e6b98.scope
    │ └─crio-9faadefe7b92202432ef1624ec22a4ec4834a74c72a308603ec7a17cfbaccdc1
      ├─crio-c7d3f699d9101a4298aefbc231d3841a812a243f29961f72a8fb4d4894c0eb31.scope
      └─crio-91209ae49c4312eee1b8f64d9e056e8b41ebde7807538f5490cb2288891be394

```

Reply:
> does this check duplicate L162?

For reference, please note that in the crio environment, use the following command to view the container process. You can see that some container progress does not have the .scope suffix.
`
cd /sys/fs/cgroup/cpuset/kubepods.slice
systemd-cgls -a

Working directory /sys/fs/cgroup/cpuset/kubepods.slice:
├─kubepods-burstable.slice
│ ├─kubepods-burstable-podbe93fa7d_608e_4c35_ad4d_1abfa5372930.slice
│ │ ├─crio-597bc6b20cc94f006b536c59ef0acd8a59559f9e36b1613c0edfc2754c04c9b3
│ │ └─crio-700d88e70d5c8afe5f7392b6ac05034ba8792b28b014dbf9a36ee6f3539fd4d3.scope
│ │   └─12180 /coredns -conf /etc/coredns/Corefile
│ ├─kubepods-burstable-pod2421dd3a_f457_470e_b5ea_fc4059b45595.slice
│ │ ├─crio-e4990d52ac5f482791ae81768e19b4fa294e59f33e12d039ec479e7036f5c966
│ │ └─crio-e2888797afa17f6f59eaf17ebc984e8f8d36c7f90e10acf3f75e747526fe3017.scope
│ │   └─10650 /bin/proxymux --config=/mnt/etc/proxymux/config.yaml --verbosity=info
│ ├─kubepods-burstable-pode36828e4_fffd_4d3f_abbc_aade0c61d7cc.slice
│ │ ├─crio-f9d22112b7038e225d4c407ced3c27c34c43db2c3b99e51fb39bfaafe26ccbb8.scope
│ │ ├─crio-9f5b9bb33cbd196637eefbd35286d5275842eef6a51d43612ef4271f2b219877
│ │ └─crio-7991c3aaa142192887ecc46656af89150dab2a4688f2265b83b462aa18adbb64.scope
│ │   └─11608 /usr/local/bin/oci-csi-node-driver --v=2 --csi-address=/csi/csi.sock --kubelet-registration-path=/var/lib/kubelet/plugins/blockvolume.csi.oraclecloud.com/csi.sock --endpoint=unix:///csi/csi.sock --nodeid=10.0.10.21 --fss-endpoint=unix:///fss/csi.sock -->
│ └─kubepods-burstable-pod9c13f043_0505_4704_85d2_850c266841d0.slice
│   ├─crio-12739bf28ed03a46b456066b757e5dce3abb343e34a365b3e853bdb90e443b17
│   └─crio-f74c1d5b7477090e8d25530d9ed41502e11d38100b74b6c8ad333823b8961c07.scope
│     └─12159 /cluster-proportional-autoscaler --namespace=kube-system --configmap=kube-dns-autoscaler --target=deployment/coredns --logtostderr=true --v=2
└─kubepods-besteffort.slice
  ├─kubepods-besteffort-pod8d5e3310_222b_4f5b_8b38_4c4dd86b0c7a.slice
  │ ├─crio-8354bd59ec2d527e7a6194cc434a633a880bb7d65927fe1093e4a55626db92da.scope
  │ ├─crio-c9037241aafce3ea26efd01134685b202bff13813a1777b67f35afa325c61f48.scope
  │ │ └─10929 /usr/local/bin/kube-proxy --config=/var/lib/kube-proxy/config.conf --hostname-override=10.0.10.21
  │ └─crio-fabe1e406efdd108fd93b2ea523103f95d4d77b661064195008c90b3dab2acd9
  └─kubepods-besteffort-pod8f06206a_e186_4875_aeda_97ef8682806b.slice
    ├─crio-9e493f2dc68a16e0e6a7980b9dd1e741949910f56dba31d88837a3b70a0ec74c.scope
    ├─crio-8eb6d44167d88f4dd5d33fa20ae218c77dc9f2acab4f6f5fdb614f7671512460.scope
    │ └─11349 /bin/oci-npn
    └─crio-75c3b9cb2681650d07ae81216ca2caaabd4226578765c4b1b42d70f04da40e99
`

Reply:
I see. I am also wondering why there are special cases having no suffixes.